subliminal/providers/cinemast.py

```diff
@@ -0,0 +1,231 @@
+
+            if r.status_code != 200:
+                raise AuthenticationError(self.username)
+
+            try:
+                result = r.json()
+                if 'token' not in result:
+                    raise AuthenticationError(self.username)
+
+                logger.info('Logged in')
+                self.user_id = r.json().get('user')
+                self.token = r.json().get('token')
+            except ValueError:
+                raise AuthenticationError(self.username)
+
+    def terminate(self):
+        # logout
+        if self.token or self.user_id:
+            logger.info('Logged out')
+            self.token = None
+            self.user_id = None
+
+        self.session.close()
+
+    def query(self, title, season=None, episode=None, year=None):
+        query = {
+            'q': title,
+            'user': self.user_id,
+            'token': self.token
+        }
+
+        # episode
+        if season and episode:
+            query['type'] = 'series'
+            query['season'] = season
+            query['episode'] = episode
+        else:
+            query['type'] = 'movies'
+            if year:
+                query['year_start'] = year - 1
+                query['year_end'] = year
+
+        # get the list of subtitles
+        logger.debug('Getting the list of subtitles')
+        url = self.server_url + 'search/'
+        r = self.session.post(url, data=query)
+        r.raise_for_status()
+
+        try:
+            results = r.json()
+        except ValueError:
+            return {}
+
+        # loop over results
+        subtitles = {}
+        for group_data in results.get('data', []):
+            # create page link
+            slug_name = group_data.get('name_en').lower().replace(' ', '-').replace('\'', '').replace('"', '')
+            if query['type'] == 'series':
+                page_link = self.server_url + 'subtitle/series/{}/{}/{}/'.format(slug_name, season, episode)
+            else:
+                page_link = self.server_url + 'subtitle/movie/{}/'.format(slug_name)
+
+            # go over each language
+            for language_code, subtitles_data in group_data.get('subtitles', {}).items():
+                for subtitle_item in subtitles_data:
+                    # read the item
+                    language = Language.fromalpha2(language_code)
+                    subtitle_id = subtitle_item['id']
+                    subtitle_key = subtitle_item['key']
+                    release = subtitle_item['version']
+
+                    # add the release and increment downloaded count if we already have the subtitle
+                    if subtitle_id in subtitles:
+                        logger.debug('Found additional release %r for subtitle %r', release, subtitle_id)
+                        bisect.insort_left(subtitles[subtitle_id].releases, release)  # deterministic order
+                        subtitles[subtitle_id].downloaded += 1
```

Comment:
Why do you increase the downloaded count? Isn't that the same as `len(subtitles[subtitle_id].releases)`?

Reply:
Done!